Lib/fontTools/misc/psCharStrings.py

```diff
@@ -251,23 +251,29 @@ class CharStringCompileError(Exception): pass
 
 class SimpleT2Decompiler(object):
 
-	def __init__(self, localSubrs, globalSubrs):
+	def __init__(self, localSubrs, globalSubrs, private):
 		self.localSubrs = localSubrs
 		self.localBias = calcSubrBias(localSubrs)
 		self.globalSubrs = globalSubrs
 		self.globalBias = calcSubrBias(globalSubrs)
+		self.private = private
 		self.reset()
 
 	def reset(self):
 		self.callingStack = []
 		self.operandStack = []
 		self.hintCount = 0
 		self.hintMaskBytes = 0
+		self.numRegions = 0
 
 	def check_program(self, program):
-		assert program, "illegal CharString: decompiled to empty program"
-		assert program[-1] in ("endchar", "return", "callsubr", "callgsubr",
-				"seac"), "illegal CharString"
+		if self.private and self.private.isCFF2:
```

Comment:
Why is this block duplicated?  It was also in an earlier commit, which I commented on.  Same issues here.